### ISSUE TITLE
Problem with sourcemaps in TypeScript project

### ISSUE THREAD

**Issue by clonardo** [react-keycloak/react-keycloak#100]
**Describe the bug**
I haven't tested this too extensively, but in a React app with the default TypeScript + Webpack configs in an Nx monorepo, I see a bunch of warnings where the sourcemaps are resolved to files that are not included in the output:

WARNING in C:/dev/1smweb/node_modules/@react-keycloak/core/lib/provider.js
Module Warning (from C:/dev/1smweb/node_modules/source-map-loader/index.js):
(Emitted value instead of an instance of Error) Cannot find source file '@react-keycloak/core/provider.tsx': Error: Can't resolve './@react-keycloak/core/provider.tsx' in 'C:\dev\1smweb\node_modules\@react-keycloak\core\lib'

/core/provider.tsx is not included in the published bundle. provider.d.ts, provider.js, and provider.js.map are, however, included. 
These are visible both in the console at build time, and this applies to every file that I have a reference to within react-keycloak/core and react-keycloak/web.

I do not have this issue with any of the (numerous other) native TypeScript libs in my project, and did not have this problem with react-keycloak until upgrading to 3.0.0 today.

**To Reproduce**

1. Scaffold a new Nx project with a React + TypeScript app
2. Install and configure `@react-keycloak/web` + `@react-keycloak/core`
3. Attempt to build
4. See errors

**Expected behavior**
Should build and work like it did before 3.0.0

**Screenshots**
N/A

**Desktop (please complete the following information):**

- OS: Windows 10
- Browser: N/A
- Version: react-keycloak/web 3.0.0, react-keycloak/core 3.0.0, Node.js 12.18.3

**Additional context**
Because I only needed a small subset of functionality from the react-keycloak project, as a temporary fix, I have just brought the few files that I needed into my build, until source mapping issues are stabilized.

**Comment by panz3r** [comment]
Hi @clonardo ,

Thanks for your feedback, I tried to reproduce the above issue using the latest versions of `nx`, `@react-keycloak/web` and `keycloak-js` but couldn't reproduce it (the `nx build` command worked as expected).

I noticed you said you installed `@react-keycloak/core` directly which is not required for the module to work (since it is a direct dependency of `@react-keycloak/web` it gets installed automatically with the right version).

Let me know if you are having more issues while using `nx`.

**Comment by clonardo** [comment]
@panz3r thanks for looking into this. I dug into this some more and found that a configuration change I had made in Nx (unrelated to `@react-keycloak` ) was actually the issue, and it just happened to not be an issue with 2.x. I've fixed things on my side, so this should be ok to close.

Thanks for your awesome work on this project!

**Comment by douglaszaltron** [comment]
@clonardo I have a similar problem, what action did you take to solve it?
I'm using webpack 5 with react-scripts 5

**Comment by douglaszaltron** [comment]
```
Compiled with problems:X

Warning:

Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from &apos;/proj/node_modules/@react-keycloak/core/lib/@react-keycloak/core/context.ts&apos; file: Error: ENOENT: no such file or directory, open &apos;/proj/node_modules/@react-keycloak/core/lib/@react-keycloak/core/context.ts&apos;

```

**Comment by tomeczekstecc** [comment]
> I still have the problem, what I did was make config-overrides.js @xyy94813
> 
> ```
> module.exports = function override(config) {
>   config.ignoreWarnings = [/Failed to parse source map/]
> ```

I had the same issue. Succeded to fix with `npm audit fix --force`

**Comment by dmoskalev** [comment]
I had the same issue. To reproduce it try:
npx create-react-app keycloak
npm install --save keycloak-js @react-keycloak/web

add ReactKeycloakProvider  to index.js:
`import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import App from './App';
import {ReactKeycloakProvider} from '@react-keycloak/web';
import Keycloak from 'keycloak-js';
const keycloak = new Keycloak();

ReactDOM.render(
  <React.StrictMode>
    <ReactKeycloakProvider authClient={keycloak}>
      <App />
    </ReactKeycloakProvider>
  </React.StrictMode>,
  document.getElementById('root')
);`

and run build. This issue only occurs with react-scripts version 5.0.0.

**Comment by georgileikin** [comment]
Same issue here. Solved it by downgrading the react-scripts version to 3.4 and typescript version to 3.9.10.
Would be nice for it to work with the react-scripts latest version.

**Comment by renefs** [comment]
This issue is still happening. CRA Typescript project:

```json
{
  "keycloak-js": "^16.1.1",
  "react-scripts": "5.0.0",
  "@react-keycloak/web": "^3.4.0",
  "typescript": "^4.5.5",
}

**Comment by edellucien** [comment]
I have the same error:
My dependencies:
`
{
  "react": "^17.0.2",
  "react-scripts": "5.0.0",
  "typescript": "^4.5.5",
  "keycloak-js": "^15.0.2",
  "@react-keycloak/web": "^3.4.0",
}`

**Comment by bsaphier** [comment]
Why is this issue closed? It's still very much an issue

**Comment by BogdanovSergey** [comment]
> @panz3r thanks for looking into this. I dug into this some more and found that a configuration change I had made in Nx (unrelated to `@react-keycloak` ) was actually the issue, and it just happened to not be an issue with 2.x. I've fixed things on my side, so this should be ok to close.
> 
> Thanks for your awesome work on this project!
> 

People are still having same annoyng messages, could you return issue status to open?

**Comment by WieFel** [comment]
+1 still the same

**Comment by keeganwhite** [comment]
Same

**Comment by MorganLmd** [comment]
Same here...

**Comment by tomeczekstecc** [comment]
...and here

**Comment by kroSiuffi2712** [comment]
same here

Failed to parse source map
module:
@react-keycloak

files:
types.ts
useKeycloak.ts
withKeycloak.tsx
provider.ts
index.ts
context.ts
provider.tsx

-----
My dependencies:
{
  "keycloak-js": "^18.0.0",
  "react-scripts": "5.0.0",
  "@react-keycloak/web": "^3.4.0",
  "typescript": "^4.5.4",
}

**Comment by minhdtb** [comment]
up. No one maintain this project ?

**Comment by mkrzywanski** [comment]
Same issue here...

**Comment by bestii** [comment]
+1

**Comment by viniciussantb** [comment]
+1

**Comment by franck-william** [comment]
+1

**Comment by sharaafnazeer** [comment]
Same here. Any solutions related to this?

**Comment by ghost** [comment]
Yes, dont use this package. Maybe, try this point of view https://github.com/dasniko/keycloak-reactjs-demo. Only one package and it works !

**Comment by t-h-e** [comment]
https://github.com/dasniko/keycloak-reactjs-demo is the right way to go. No need for `react-keycloak`. `keycloak-js` is sufficent and actually my code has become simpler.

**Comment by toky03** [comment]
I have the same Issue

**Comment by dyluc** [comment]
It's not a solution, but you can change the webpack config to ignore this warning...

1. Install [react-app-rewired](https://github.com/timarney/react-app-rewired) with:
`npm install react-app-rewired --save-dev`
2. Create a `config-overrides.js` in your project root dir:
```js
module.exports = function override(config, env) {
    config.ignoreWarnings = [/Failed to parse source map/]
    return config;
}
```
3. Update the scripts in `package.json`:
```js
"scripts": {
    "start": "react-app-rewired start",
    "build": "react-app-rewired build",
    "test": "react-app-rewired test",
    "eject": "react-scripts eject"
},
```

Restart and you should not see the warnings.

**Comment by agmCorp** [comment]
not work for me

**Comment by houssemamiri-logient** [comment]
Does any one have solutiin ?

**Comment by RMCampos** [comment]
Same here. Is not possible that no one is taking a look on how to get this fixed!

**Comment by RMCampos** [comment]
Found the open issue, in case of someone get interested in: #176

**Comment by nrl240** [comment]
According to CRA, sourcemaps should be fixed in this package, as the errors/warnings stem from the package.
See https://github.com/facebook/create-react-app/discussions/11767#discussioncomment-1969071

https://github.com/facebook/create-react-app/discussions/11767#discussioncomment-2245392
> You need to include source code in your npm package for source maps to work. If you don't want to do that you should probably stop building source maps as well

**Comment by gnowland** [comment]
I submitted a PR to fix this #200
